### PR TITLE
 Add docker release to the full release process for final releases

### PR DESCRIPTION
**Pull request by mikealfare**
### Problem

Publishing a Docker release is currently a manual process which can be forgotten.

### Solution

Add the Docker release to the release pipeline.

### Checklist

- [x] I have read [the contributing guide](https://github.com/dbt-labs/dbt-core/blob/main/CONTRIBUTING.md) and understand what's expected of me
- [x] I have run this code in development and it appears to resolve the stated issue
- [x] This PR includes tests, or tests are not required/relevant for this PR
- [x] This PR has no interface changes (e.g. macros, cli, logs, json artifacts, config files, adapter interface, etc) or this PR has already received feedback and approval from Product or DX

### CODE REVIEW

File: .github/workflows/release.yml [hunk 1]
```diff
@@ -195,34 +184,38 @@ jobs:
...
       PYPI_API_TOKEN: ${{ secrets.PYPI_API_TOKEN }}
       TEST_PYPI_API_TOKEN: ${{ secrets.TEST_PYPI_API_TOKEN }}
 
+  docker-release:
+    name: "Docker Release"
+    if: ${{ !failure() && !cancelled() }}
+    needs: [bump-version-generate-changelog, build-test-package, github-release]
+    permissions:
+      packages: write
+    uses: dbt-labs/dbt-release/.github/workflows/release-docker.yml@docker-release
```
Comment (mikealfare) [inline .github/workflows/release.yml:202]: Point this back to main

**Comment by mikealfare** [comment]
Test release using this branch: https://github.com/dbt-labs/dbt-bigquery/actions/workflows/release.yml

File: .github/workflows/nightly-release.yml [hunk 2]
```diff
@@ -20,6 +20,7 @@ on:
 
 permissions:
   contents: write # this is the permission that allows creating a new release
+  packages: write # this is the permission that allows Docker release
```
Comment (VersusFacit) [inline .github/workflows/nightly-release.yml:23]: nit: allows what to Docker release :p

File: docker/test.sh [hunk 3]
```diff
@@ -0,0 +1,19 @@
+# - VERY rudimentary test script to run latest + specific branch image builds and test them all by running `--version`
```
Comment (VersusFacit) [inline docker/test.sh:1]: remove script

**Review by VersusFacit — APPROVED** [review]

**Pull request merged**